denokv/main.rs

```diff
@@ -344,7 +352,9 @@ async fn metadata_endpoint(
   let Some(Json(req)) = maybe_req else {
     return Err(ApiError::MinumumProtocolVersion);
   };
-  if !req.supported_versions.contains(&2) {
+  if !req.supported_versions.contains(&2)
+    && !req.supported_versions.contains(&3)
+  {
```

Comment:
Can you return the negotiated version in the metadata response below?

Something like:
```rust
  let version = if req.supported_versions.contains(&3) { 3 } else { 2 };
  Ok(Json(DatabaseMetadata {
    version,
```